src/components/integrated/OakQuizCheckBox/OakQuizCheckBox.tsx

```diff
@@ -157,14 +157,20 @@ export const OakQuizCheckBox = (props: OakQuizCheckBoxProps) => {
     disabled,
     innerRef,
     displayValue,
+    defaultChecked,
     ...rest
   } = props;
 
   const isFeedback = !!feedback;
-  const isCorrect = feedback === "correct";
 
   const defaultRef = useRef<HTMLInputElement>(null);
   const inputRef = innerRef ?? defaultRef;
+  const showTick =
+    (feedback === "correct" && (inputRef.current?.checked || defaultChecked)) ||
+    (feedback === "incorrect" && !inputRef.current?.checked && !defaultChecked);
+  const showCross =
+    feedback === "incorrect" && (inputRef.current?.checked || defaultChecked);
+
```

Comment:
My gut is telling me there is a bug in here. Do you need to reference `defaultChecked`, is it enough to read the `checked` state from the ref?

I reckon this component could get in a state where the feedback is out of sync with the checkbox. `defaultChecked` is the initialising state for the checkbox but you're always referencing it here so in some cases it will trump the current `checked` state of the checkbox. 

If you can remove `defaultChecked` altogether I would. Failing that i'd make the internal checkbox controlled and manage its checked state with `useState` .

Something like:

```typescript
const [isChecked, setIsChecked] = useState(defaultChecked);
const showTick =
    (feedback === "correct" && isChecked) ||
    (feedback === "incorrect" && !isChecked);
  const showCross =
    feedback === "incorrect" && isChecked);
const handleChange: ChangeEventHandler<HTMLInputElement> = (event) => {
  setIsChecked(event.currentTarget.checked);
  if (onChange) onChange(event);
}
```

More code, but it sees off a subtle bug that would be hard to identify down the road. Hope that makes sense.

Reply:
It makes sense but changes so much of the implementation higher up that I really don't want to do it at this stage. DefaultChecked is basically only used for the StoryBook so I'm wondering whether the solution is just to not expose and use defaultChecked at all.

Reply:
Which is of course what you said 🙄